tools/mock_node/src/main.rs

```diff
@@ -12,57 +15,67 @@ use near_crypto::{InMemorySigner, KeyType};
-/// The simulated peers stay at height 100.
 ///
-/// start_mock_node ~/.near/localnet/node0 -s 61 -h 100 --mode produce_new_blocks
+/// Examples
 ///
-/// Both client and mock network starts at height 61, mock network will produce new blocks until height 100
-///  
-/// start_mock_node ~/.near/localnet/node0 -h 100 --mode "produce_new_blocks(20)"
+/// ```console
+/// # Pure catchup from genesis height to the end of the recorded history.
+/// $ mock-node ~/.near/localnet/node0
 ///
-/// Client starts at genesis height and mock network starts at heigh 20,
-/// mock network will produce new blocks until height 100
+/// # Pure block production starting from block height 61.
+/// $ mock-node ~/.near/localnet/node0 --start-height 61
+///
+/// # Mixed: client starts at genesis and tries to catch up with the network, which starts at height 20.
+/// $ mock-node ~/.near/localnet/node0 --network-height 20
+/// ```
 #[derive(Parser)]
 struct Cli {
     /// Existing home dir for the pre-generated chain history. For example, you can use
     /// the home dir of a near node.
     chain_history_home_dir: String,
     /// Home dir for the new client that will be started. If not specified, the binary will
     /// generate a temporary directory
-    client_home_dir: Option<String>,
+    client_home_dir: Option<PathBuf>,
     /// Simulated network delay (in ms)
     #[clap(short = 'd', long, default_value = "100")]
     network_delay: u64,
-    /// Mode of the mock network, choices: [no_new_blocks, produce_new_blocks, produce_new_blocks(u64)]
-    #[clap(short = 'M', long)]
-    mode: MockNetworkMode,
-    /// If specified, the binary will set up client home dir before starting the client node
-    /// so head of the client chain will be the specified height when the client starts.
-    /// The given height must be the last block in an epoch.
-    #[clap(short = 's', long)]
-    client_start_height: Option<BlockHeight>,
+    /// If specified, the binary will set up client home dir before starting the
+    /// client node so head of the client chain will be the specified height
+    /// when the client starts. The given height must be the last block in an
+    /// epoch.
+    #[clap(long, default_value = "0")]
+    client_height: BlockHeight,
+    /// The height at which the mock network starts. The client would have to
+    /// catch up to this height before participating in new block production.
+    ///
+    /// Defaults to the largest height in history.
+    #[clap(long)]
+    network_height: Option<BlockHeight>,
+    /// Shortcut to set both `--client-height` and `--network-height`.
+    #[clap(long, conflicts_with_all(&["client-height", "network-height"]))]
+    start_height: Option<BlockHeight>,
     /// Target height that the client should sync to before stopping. If not specified,
     /// use the height of the last block in chain history
-    #[clap(short = 'h', long)]
+    #[clap(long)]
```

Comment:
Note: I intentionally left out short options from here. They are an anti-pattern: for rarely used utilities, the reader can benefit a lot from seeing `--target-height 92` rather than `-h 92`